lib/BackbeatProducer.js

```diff
@@ -68,7 +68,14 @@ class BackbeatProducer extends EventEmitter {
             'request.timeout.ms': ACK_TIMEOUT,
         });
         this._ready = false;
-        this._producer.connect();
+        this._producer.connect({ timeout: 30000 }, () => {
+            const opts = { topic: 'backbeat-sanitycheck', timeout: 10000 };
```

Comment:
Maybe you could get metadata from `this._topic` instead of using a dummy topic?

Reply:
I think calling `getMetadata` on a topic creates it if does not exist, which could possibly be avoided by using an existing topic.

Reply:
This would create the `backbeat-replication` topic with some partition, replica defaults apart from what's specified in the config, which is not ideal.

Reply:
From the library docs: 
Note: using a metadataOptions.topic parameter has a potential side-effect. A Topic object will be created, if it did not exist yet, with default options and it will be cached by librdkafka. A subsequent call to create the topic object with specific options (e.g. acks) will return the previous instance and the specific options will be silently ignored. To avoid this side effect, the topic object can be created with the expected options before requesting metadata, or the metadata request can be performed for all topics (by omitting metadataOptions.topic).

Should we reserve a topic name for the specific purpose of checking if kafka is available? I think crashing is useful as well for non-kubernetes use instead of hanging

Reply:
Good Idea!

Reply:
@rahulreddy I don't think we have code to pre-create topics for now in kube stack, but when we do it it should not be an issue anymore. And I'm thinking we need this setup code in order to create topics with the number of partitions and replicas that we want?

Reply:
Unless we let kube do the topic setup at the same time it attempts to start other components (that seems to be the philosophy of it?) in which case indeed it becomes an issue to let backbeat auto-create topics.

Reply:
@jonathan-gramain Using ```this._topic``` also makes CI fail for some tests, unlike using the ```backbeat-sanitycheck``` topic.